api/client/ffs.go

```diff
@@ -572,7 +573,7 @@ func protoToDealError(des []*rpc.DealError) ([]ffs.DealError, error) {
 			var err error
 			propCid, err = cid.Decode(de.ProposalCid)
 			if err != nil {
-				propCid = cid.Undef
+				return nil, fmt.Errorf("proposal cid is invalid")
```

Comment:
This was interesting. It always returned nil error.
@asutula , I thought this would be a better decision, agree?

Reply:
Yes, I like that. Think I copy/pasted some code from the lotus cli here. Your solution is what I would have don't if I typed it myself or noticed this before ;)